src/Nethermind/Nethermind.Core/ByteArrayConverter.cs

```diff
@@ -0,0 +1,114 @@
+// SPDX-FileCopyrightText: 2022 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System;
+using System.Buffers;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.CompilerServices;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Nethermind.Core.Extensions;
+
+namespace Nethermind.Serialization.Json;
+
+public class ByteArrayConverter : JsonConverter<byte[]>
+{
+    public override byte[]? Read(
+        ref Utf8JsonReader reader,
+        Type typeToConvert,
+        JsonSerializerOptions options)
+    {
+        return Convert(ref reader);
+    }
+
+    public static byte[]? Convert(ref Utf8JsonReader reader)
+    {
+        JsonTokenType tokenType = reader.TokenType;
+        if (tokenType == JsonTokenType.None || tokenType == JsonTokenType.Null)
+        {
+            return null;
+        }
+        else if (tokenType != JsonTokenType.String)
+        {
+            ThrowInvalidOperationException();
+        }
+
+        int length = reader.ValueSpan.Length;
+        byte[]? bytes = null;
+        if (length == 0)
+        {
+            length = checked((int)reader.ValueSequence.Length);
+            if (length == 0)
+            {
+                return null;
+            }
+
+            bytes = ArrayPool<byte>.Shared.Rent(length);
+            reader.ValueSequence.CopyTo(bytes);
+        }
+
+        ReadOnlySpan<byte> hex = bytes is null ? reader.ValueSpan : bytes.AsSpan(0, length);
+        if (hex.StartsWith("0x"u8))
+        {
+            hex = hex[2..];
+        }
+
+        byte[] returnVal = Bytes.FromUtf8HexString(hex);
+        if (bytes is not null)
+        {
+            ArrayPool<byte>.Shared.Return(bytes);
+        }
```

Comment:
put in finally?

Reply:
`ArrayPool<byte>.Shared` doesn't really care if you return the buffer; it just means higher allocation from re-creating a new one; however try,finally does add overhead so is more optimal to not return in finally if a try block doesn't already exist and chance of failure is low.